firestore-palm-summarize-text/functions/src/generator.ts

```diff
@@ -54,33 +59,101 @@ export class TextGenerator {
     this.candidateCount = options.candidateCount;
     this.instruction = options.instruction;
     if (options.model) this.model = options.model;
-    logs.usingADC();
-    const auth = new GoogleAuth({
-      scopes: [
-        'https://www.googleapis.com/auth/userinfo.email',
-        'https://www.googleapis.com/auth/generative-language',
-      ],
-    });
-    this.client = new TextServiceClient({
-      auth,
-    });
+
+    this.endpoint = `projects/${config.projectId}/locations/${config.location}/publishers/google/models/${this.model}`;
+
+    if (config.useVertex) {
```

Comment:
If we use provider naming, this could be extracted out into `providers/models`. 

The implementation could then be loaded from that file?

models/vertex
models/generativeai

As examples.